okta/users.go

```diff
@@ -90,34 +106,49 @@ func resourceUserRead(d *schema.ResourceData, m interface{}) error {
 
 	userList, _, err := client.Users.GetByID(d.Get("email").(string))
 	if err != nil {
-		// if the user does not exist in okta, delete from terraform
-		// results in a terraform plan to create the user
+		// if the user does not exist in okta, delete from terraform config
 		if client.OktaErrorCode == "E0000007" {
 			d.SetId("")
 			return nil
 		} else {
 			log.Printf("[ERROR] Error GetByID: %v", err)
 			return err
 		}
+	} else {
+		userRoles, _, err := client.Users.ListRoles(userList.ID)
+		if err != nil {
+			log.Printf("[ERROR] Error listing user role: %v", err)
+			return err
+		}
+		if userRoles != nil {
+			if len(userRoles.Role) > 1 {
+				log.Printf("[ERROR] User has more than one role. This terraform provider presently only supports a single role per user. Please review the role assignments in Okta for this user.")
+				return fmt.Errorf("User has more than one role. This terraform provider presently only supports a single role per user. Please review the role assignments in Okta for this user.")
```

Comment:
Do we need both of these print lines?

Reply:
its consistent with how errors are bring handled in the provider. errors are logged and also returned. it may look odd as this is the only time (so far) i'm creating the error message instead of just returning the error message produced by the okta sdk.

when i have a better understanding of go logging i can see abstracting this into a function.